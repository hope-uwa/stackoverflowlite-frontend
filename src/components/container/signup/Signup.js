/* eslint-disable */
import React, { Component } from 'react';
import { ToastContainer } from "react-toastify";


class Signup extends Component {
  constructor(props) {
    super(props);
    
    this.state ={
        email: '',
        username: '',
        password: '',
        signupDisplay: 'none'
    }
  }

  closeModal = ()=>{   
    this.setState({
        email: '',
      username: '',
      password: '',
        signupDisplay: 'none'
    })
  }
  componentWillReceiveProps =(props) =>{
     const display = props.signupDisplay
    this.setState({ signupDisplay:display });
      
  }
  

  signupSubmit = async (event) => {
    event.preventDefault();
    const userData = { ...this.state };
    event.persist()
    event.target.innerHTML= 'LOADING......'
    const action = await this.props.signup(userData);
    event.target.innerHTML= 'SIGN UP'
    if(this.props.signupInfo.token){
    setTimeout(()=>this.setState({signupDisplay:'none'}), 1500)
    
    }
  }

  handleChange= (event) => {
    this.setState({
      [event.target.id]: event.target.value
    });
  }

  render() {
    const display= this.state.signupDisplay;
    const style ={
        display,
    }
    const {signup} = this.props
    return (
            <div id="signUpModal" className="modal" style={style}>
                <div className="modal-content">
                    <div className="modal-head w100">
                        <span className="pl1  modal-title ">Join Our Community!</span>
                        <span id="s-close" onClick={this.closeModal} className="close">&times;</span>
                    </div>
                    <form id="signup-form" className="mt2" onSubmit={this.signupSubmit}>
                        <div className="modal__signup-form row">             
                                <div className="modal-form align-center">                  
                                <input type="text" required placeholder="Enter Your Nickname" id="username" className="" name="username" onChange={this.handleChange}/>
                                <input type="email" required placeholder="Enter Email Address" id="email" className="" name="email" onChange={this.handleChange} />
                                <input type="password" required placeholder="Enter Password" id="password" className="" name="password" onChange={this.handleChange} />                   
                                </div>            
                        </div>          
                        <div className="modal-footer  w-100 pt2 pb2">             
                                <button type="submit" onClick={this.signupSubmit} className="btn btn-shadow btn_login-signup w60">SIGN UP</button>
                        </div>
                    </form>
                </div>
                <ToastContainer autoClose={2000} />
            </div>
    );
  }
}

export default Signup;