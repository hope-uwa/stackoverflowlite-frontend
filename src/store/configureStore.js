import { createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import rootReducer from '../reducers';

const middleware = process.env.NODE_ENV !== 'production' ?
  [require('redux-immutable-state-invariant').default(), thunk] :
  [thunk];

function configureStore(initialState) {
  return createStore(
    rootReducer,
    initialState,
    applyMiddleware(...middleware),
    );
}
const store = configureStore();
window.store = store;
store.subscribe(()=> localStorage.setItem('user',JSON.stringify(store.getState().loginReducer)));
export default store;