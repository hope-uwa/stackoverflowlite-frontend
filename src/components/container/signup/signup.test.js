/* eslint-disable no-undef */
import React from 'react';
import { mount, shallow } from 'enzyme';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import Signup from './Signup';


const middlewares = [thunk];
const mockStore = configureStore(middlewares);


describe('<Signup />', () => {
  it('matches the snapshoot', () => {
    const initialState = {};
    const store = mockStore(initialState);

    const wrapper = shallow(<Provider store={store}>
      <Signup/>
    </Provider>);
    expect(wrapper).toMatchSnapshot();
  });


  it('render sigup page correctly', () => {
    const wrapper = mount(<Signup />);
    expect(wrapper.find('button').text()).toBe('SIGN UP');
    expect(wrapper.find('.modal-title').text()).toBe('Join Our Community!');
    expect(wrapper.find('form input').length).toEqual(3);
  });

  it('should update states when data is passed', () => {
    const wrapper = mount(<Signup />);
    const emailInput = wrapper.find('#email');
    emailInput.instance().value = 'uwahope007@gmail.com';
    emailInput.simulate('change');

    const usernameInput = wrapper.find('#username');
    usernameInput.instance().value = 'uwahope007';
    usernameInput.simulate('change');

    const passwordInput = wrapper.find('#password');
    passwordInput.instance().value = 'password123';
    passwordInput.simulate('change');

    expect(wrapper.state()).toMatchObject({
      email: 'uwahope007@gmail.com',
      username: 'uwahope007',
      password: 'password123',
    });
    wrapper.unmount();
  });

  it('should submit inputs from the state', () => {
    const mockSignupFn = jest.fn();
    const wrapper = mount(<Signup signup={mockSignupFn} />);

    const emailInput = wrapper.find('#email');
    emailInput.instance().value = 'anasey@outlook.com';
    emailInput.simulate('change');

    const usernameInput = wrapper.find('#username');
    usernameInput.instance().value = 'anasey001';
    usernameInput.simulate('change');

    const passwordInput = wrapper.find('#password');
    passwordInput.instance().value = 'password123';
    passwordInput.simulate('change');

    const form = wrapper.find('form');
    form.simulate('submit');
    expect(mockSignupFn.mock.calls.length).toBe(1);
    wrapper.unmount();
  });
});
