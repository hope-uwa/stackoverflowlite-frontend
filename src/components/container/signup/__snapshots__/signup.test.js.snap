// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Signup /> matches the snapshoot 1`] = `<Signup />`;
